src/FixedDataTableBufferedRows.js

```diff
@@ -77,119 +77,96 @@ class FixedDataTableBufferedRows extends React.Component {
   }
 
   render() /*object*/ {
-    var props = this.props;
-    var rowClassNameGetter = props.rowClassNameGetter || emptyFunction;
-    var rowsToRender = this.props.rowsToRender || [];
-
-    if (props.isScrolling) {
-      // We are scrolling, so there's no need to display any rows which lie outside the viewport.
-      // We still need to render them though, so as to not cause any mounts/unmounts.
-      this._staticRowArray.forEach((row, i) => {
-        const rowOutsideViewport = !this.isRowInsideViewport(row.props.index);
-        if (rowOutsideViewport) {
-          this._staticRowArray[i] = this.getStubRow(i, false, row.props.index);
-        }
-      });
+    let { offsetTop, scrollTop, isScrolling, rowsToRender } = this.props;
+    const baseOffsetTop = offsetTop - scrollTop;
+    rowsToRender = rowsToRender || [];
+
+    if (isScrolling) {
+      // allow static array to grow while scrolling
+      this._staticRowArray.length = Math.max(this._staticRowArray.length, rowsToRender.length);
     } else {
+      // when scrolling is done, static array can shrink to fit the buffer
       this._staticRowArray.length = rowsToRender.length;
     }
 
-    var baseOffsetTop = props.offsetTop - props.scrollTop;
-
-    for (let i = 0; i < rowsToRender.length; i++) {
-      const rowIndex = rowsToRender[i];
-
-      // if the row doesn't exist in the buffer, assign a fake row to it.
-      // this is so that we can get rid of unnecessary row mounts/unmounts
+    // render each row from the buffer into the static row array
+    for (let i = 0; i < this._staticRowArray.length; i++) {
+      let rowIndex = rowsToRender[i];
+      // if the row doesn't exist in the buffer set, then take the previous one
       if (rowIndex === undefined) {
-        // if a previous row existed, let's just make use of that
-        if (this._staticRowArray[i] === undefined) {
-          this._staticRowArray[i] = this.getStubRow(i, true, -1);
-        }
-        continue;
+        rowIndex = this._staticRowArray[i] && this._staticRowArray[i].props.index;
```

Comment:
here we take the previous rendered row to avoid unmounts

if it is still undefined (aka, fake row), it will be rendered as null